### ISSUE TITLE
cmd/go: list all GOARCH/GOOS

### ISSUE THREAD

**Issue by jimmyfrasche** [golang/go#27916]
It's sometimes useful for tools to know about GOARCH/GOOS other than the current one. This information has to be collected from various sources, hardcoded, and manually kept up to date. It would be better if the go command could be queried to find out what the current information is.

Ideally, I'd like to see something roughly like the below, with the option to have it be output as json.

```
struct {
  Supported []struct {
    GOOS string
    GOARCH string
    CgoSupported bool
    WordSize int64
    MaxAlign int64
  }
  ReservedGOARCH []string
  ReservedGOOS []string
}
```

cc: @bcmills @rsc @ianlancetaylor @alandonovan

**Comment by dominikh** [comment]
@jimmyfrasche Is `go tool dist list` what you're looking for?

**Comment by bontibon** [comment]
Duplicate issue that was already fixed: #12270

**Comment by jimmyfrasche** [comment]
Ooh, I did not know about that one. It is indeed. Thank you.
